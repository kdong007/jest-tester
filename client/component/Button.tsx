import React from 'react';

const Button: React.FC = () => <button type="button">Button</button>;

export default Button;
