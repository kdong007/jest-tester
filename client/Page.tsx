import Button from '@component/Button';
import React from 'react';

const Page: React.FC = () => (
  <div>
    <Button />
    <Button />
  </div>
);

export default Page;
