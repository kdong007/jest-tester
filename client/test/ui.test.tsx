import Button from '@component/Button';
import React from 'react';
import renderer from 'react-test-renderer';

import Page from '../Page';

test('Button test', () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('page test', () => {
  const tree = renderer.create(<Page />).toJSON();
  expect(tree).toMatchSnapshot();
});
