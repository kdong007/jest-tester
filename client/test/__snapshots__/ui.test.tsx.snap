// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test 1`] = `
<button
  type="button"
>
  Button
</button>
`;

exports[`page test 1`] = `
<div>
  <button
    type="button"
  >
    Button
  </button>
  <button
    type="button"
  >
    Button
  </button>
</div>
`;
